Turn `rpc::Client::new` into a sync fn by not performing the health check on creation

romac opened the pull request:
See #168 for the rationale behind this change.

In addition, both methods now take ownership of the `rpc::Address` rather than cloning a reference. This avoids unnecessary clones when the caller already has ownership of the address.

<!--

Thanks for filing a PR! Before hitting the button, please check the following items.
Please note that every non-trivial PR must reference an issue that explains the
changes in the PR.

-->

* [x] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [x] Wrote tests
* [ ] Updated CHANGES.md

tarcieri commented:
Yeah, I do remember this feeling weird when I `async`-ified it. It's nice for constructors to be synchronous, I think.

romac commented:
My bad, sorry.

Alright, I removed the `new_healthy` method, fixed the integration tests, and added a new test for the `/healthy` endpoint.

Unfortunately, some integration tests fail for me when running them with `cargo test -- --ignored` against `tendermint node --proxy_app=kvstore`(Tendermint 0.33.1-73c19bd6, latest master):

```
test rpc::health ... ok
test rpc::abci_query ... ok
test rpc::block_results ... FAILED
test rpc::abci_info ... FAILED
test rpc::genesis ... FAILED
test rpc::net_info ... ok
test rpc::block ... FAILED
test rpc::commit ... FAILED
test rpc::blockchain ... FAILED
test rpc::status_integration ... ok
```

<details>
<summary><b>Click here for the full log</b> </summary>

```
failures:

---- rpc::block_results stdout ----
thread 'rpc::block_results' panicked at 'called `Result::unwrap()` on an `Err` value: Error { code: ParseError, message: "Parse error. Invalid JSON", data: Some("missing field `results` at line 11 column 3") }', src/libcore/result.rs:1188:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.

---- rpc::abci_info stdout ----
thread 'rpc::abci_info' panicked at 'called `Result::unwrap()` on an `Err` value: Error { code: ParseError, message: "Parse error. Invalid JSON", data: Some("parse error at line 11 column 5") }', src/libcore/result.rs:1188:5

---- rpc::genesis stdout ----
thread 'rpc::genesis' panicked at 'called `Result::unwrap()` on an `Err` value: Error { code: ParseError, message: "Parse error. Invalid JSON", data: Some("missing field `max_age` at line 17 column 9") }', src/libcore/result.rs:1188:5

---- rpc::block stdout ----
thread 'rpc::block' panicked at 'called `Result::unwrap()` on an `Err` value: Error { code: ParseError, message: "Parse error. Invalid JSON", data: Some("missing field `num_txs` at line 37 column 7") }', src/libcore/result.rs:1188:5

---- rpc::commit stdout ----
thread 'rpc::commit' panicked at 'called `Result::unwrap()` on an `Err` value: Error { code: ParseError, message: "Parse error. Invalid JSON", data: Some("missing field `num_txs` at line 37 column 7") }', src/libcore/result.rs:1188:5

---- rpc::blockchain stdout ----
thread 'rpc::blockchain' panicked at 'called `Result::unwrap()` on an `Err` value: Error { code: ParseError, message: "Parse error. Invalid JSON", data: Some("missing field `num_txs` at line 40 column 9") }', src/libcore/result.rs:1188:5


failures:
    rpc::abci_info
    rpc::block
    rpc::block_results
    rpc::blockchain
    rpc::commit
    rpc::genesis

test result: FAILED. 4 passed; 6 failed; 0 ignored; 0 measured; 0 filtered out

Liamsi commented:
Failing integration tests could be related to #95

Liamsi commented:
Yeah, all these `Parse error. Invalid JSON` are because the RPC endpoints changed in tendermint (go) and we didn't catch up with the changes. #120 will help us to keep up with those.

romac commented:
Just updated the PR title.

Liamsi commented:
If you want to be sure you didn't introduce a new problem, you can re-check the integration tests against 0.32.7 (together with the changes from #95 <- we should merge that one too).
